### ISSUE TITLE
performance-inefficient-vector-operation false-negatives when looping on multiple vectors

### ISSUE THREAD

**Issue by nick-potenski** [llvm/llvm-project#179458]
When a loop performs inefficient operations on multiple vectors, performance-inefficient-vector-operation does not emit a warning.

```c++
#include <vector>

int main() {
    std::vector<int> bob;
    std::vector<float> tom;

    for (auto x = 0; x < 100; x++) {
        bob.emplace_back(x);
        tom.push_back(x);
    }
}
``` 

does not emit any warnings.  However, commenting out one or the other vectors from the same code does:

```c++
#include <vector>

int main() {
    // std::vector<int> bob;
    std::vector<float> tom;

    for (auto x = 0; x < 100; x++) {
        // bob.emplace_back(x);
        tom.push_back(x);
    }
}
``` 

```
<source>:9:9: warning: 'push_back' is called inside a loop; consider pre-allocating the container capacity before the loop [performance-inefficient-vector-operation]
    7 |     for (auto x = 0; x < 100; x++) {
    8 |         // bob.emplace_back(x);
    9 |         tom.push_back(x);
      |         ^
1 warning generated.
```

**Comment by frederick-vs-ja** [comment]
I guess the crux is that Clang-tidy is confused by interleaving side effects.
